### DISCUSSION TITLE
DSC Pc 1616 bricked?

### DISCUSSION THREAD

**Discussion by BigHomie90** [taligentx/dscKeybusInterface#169] [dscKeybusInterface Discussion]
I know that is NOT something related with your code but i wanted to post here in case someone of you guys  faced this before.
The alarm was working ok and i was expirementing with it. I try to remeber what i did change the last time and now its not working but i don't remember...On boot relay its clicking 10 times but its not for the installer lockout because it doesnt do it once. It does this inifinite times. Ii looks like it is bricked.. Not completing boot. Please someone if can verify this, tell me and i ll buy a new one.

https://youtu.be/NFDMC2Sc_SI

**Comment by Dilbert66** [comment]
Did you try booting with only the keypad connected? This could also be an issue with noise on the bus or bad connection .

You can also try and default the system (reset to factory)
https://www.alarmgrid.com/faq/how-do-i-default-my-dsc-powerseries-alarm-panel

**Reply by BigHomie90** [reply]
I tried with and without the keypad.. Also tried to reset to factory using the jumper multiple times..
I googled about this a LOT but i start to believe that i am the first one who achieved to brick the board cause i didnt find anywhere this issue.. Just the typical 10 clicks from the relay for installer lockout.. Not infinite loop like my case...

**Reply by Dilbert66** [reply]
Interesting. I'm out of ideas!  Well, last check. I assume your adapter is putting out approx 16vac...

**Reply by BigHomie90** [reply]
AC Voltage 17.8 - 17.9.

Me too unfortunately.. Thank you for replying!

**Comment by taligentx** [comment]
Hi @BigHomie90,

>On boot relay its clicking 10 times but its not for the installer lockout because it doesnt do it once. It does this inifinite times. Ii looks like it is bricked

Unfortunately I haven't seen this behavior either, what is the DC voltage at Aux +/-?  Does the voltage change at all while the panel reboots?

**Reply by BigHomie90** [reply]
If I remember correctly aux output was 12.6 or 13.6v
The last thing I did to check the board was to run the Keybus reader sketch (Log: Unknown data CRC=OK I think it wrote multiple times. My bad I didn't keep the log). And one more time using status sketch( Log: Keybus connected, Keybus disconnected,Keybus connected, Keybus disconnected infinite loop). So I throwed away this board and am I waiting for a new pc1616 in about a week. I ll happily help with your sketches testing etc.

**Comment by kricon** [comment]
Did you have installer lockout enabled before panel 'died'?
It looks like panel is starting normally, with installed lockout enabled, but then restart itself (only logic, not output voltages). Like it stuck in bootloop for unknown reason, it can be like shorted output or possibly corrupted firmware. Maybe corruption enabled installer lockout, i dont know - i remember older classic series panels can often corrupt programming if panel continue to run on failing battery (under 10V).
If I remember corectly, I saw someone had problem similar to yours, but on his panel relay was clicking non-stop, without pause like yours, and it was older powerseries panel. Edit: found it: https://www.youtube.com/watch?v=EnmRYDaXWvo

I suggest you to use battery always connected to the panel, it can give more current to panel if needed (which should not be case with yours, as you have only one keypad and I suppose two wemos connected onto it). Without it you will see little led flickering on keypad every 10 seconds or so when panel is checking for battery. You can even start panel just from battery by momentairy connecting negative side from battery to any AC terminal, panel will power up fine even without AC power using that trick so you can verify if its transformer problem.

BTW: as you hold # key for 2 seconds, keypad itself goes into "select partition" menu, and then you should hold key 1 (relative to active partion) for 2 seconds to select on which partition keypad operates. Long beeps and blank keypad means keybus error.

Just of curiosity, is it by any way possible this is related to `dsckeybusinterface`? For example, panel was 'dying' and that is possible reason why unlocker sketch failed multiple times? Or maybe it is possible to send faulty data to keybus which can corrupt programming and brick the panel? I have some panels including PC1832 ready, if needed, to test `unlocker` sketch from `expander` branch.

**Reply by BigHomie90** [reply]
No installer lockout was disabled. I tried many things before tossing the board. (connecting a good health battery, pressing # for 2 seconds and then 1 on the keypad, short circuit the board etc). The last things i did to the board while it was working was to connect to it using dls and program some zones. First i downloaded the data to the pc and then upload them to the panel. After that i made some adjustments on the keypad function buttons. *8 + installer code + 000 and then the sections for funtion keys. After that the physical keypad went unresponsive so i power recycled the board.... Guess what? It never booted ever since :P

*** I used the unlocker sketch (expander) on a same panel (pc1616) that my parents have and it behave like the one i bricked when it found the installer code. So i don't think that it was my boards specific problem.

One thing to mention. The wiring is as the diagram shows but the NPN trnasistor i used is a TIP122. It's pinout order (base,collector,emmiter) is different from a other NPN's. So i wired it according to TIP122.

I am starting to believe that something was messed up when using the dls.. Maybe the connection didnt terminate as it should...But the thing is after the dls i could still access the system from the physical keybus and make changes...As i did on section 000.

**Reply by kricon** [reply]
I have found video about other guy having issue similar to yours and edited the post above, give it a look.

How do you connect PC-Link to RS232? Directly to COM port of pc/laptop or thru some usb-serial adapters? I dont belive that it could be the problem (RS232 can have voltages up to 25V AFAIK), I have programmed my PC1616 panel thru PC-Link and old Pentium3 laptop, everything works like charm, system is installed at my weekend house - when I go there, I can check `unlocker` sketch on PC1616. I even reversed PC-Link connector rx/tx multiple times, no issues, and I have program dozen of panels thru DLS and homemade cable.

If nothing helps, before tossing the board I would check the board using multimeter, for any burned diodes, shorts, bad transistors, or broken traces. At last you can even try physically wiggling and bending the board - as its multilayer maybe there is short somewhere between layers. I know it sounds weird but I had issue with PC5516 keypad where Zone 14 and program led was contrantly lit but if I bend the board or freeze it, it would work for some time. Lastly I desolder transisors driving led coumn and rows, but led was still lit (althrough disconnected from main IC!) and I found a short between PCB layers near program leds - fixed it and keypad works fine. On another PC5508 keypad I was experimenting to transform it into 32 zone, last row of led wasnt working and I have found one broken thru-hole via, fixed by soldering a jumperwire.

**Reply by BigHomie90** [reply]
Because i don't have pc link cable, i used jumper cables from mainboard to a serial port directly on a dell optiplex 990.(Funny but it worked straight away).  
The two middle pins on the board "bridged" and on the other end connected to serial port pin number 5. The other two jumper wires (one on  serial pin 2 and the other one on serial pin 3) Something like this video ---> https://www.youtube.com/watch?v=tkKrRc5Mysg&t=72s.I already tossed the board :( ... Basically i went nuts with the clicking noise going on and on so first i broke it and then throw it away...Believe me, it doesnt seem weird to me cause i had the same issue (short somewhere between layers) but on pc motherboard! i didnt try this the truth is..Anyways.. I was just searching the reason that the board died in order to not brick the next one i am waiting to recieve...I saw the video you posted and someone said at commments something about EEPROM. But how could i brick it? i wasnt doing an upgrade.. I was just adjusting some specific settings..Even if i accidentally enabled installer lockout it wouldnt behave like that..

**Reply by kricon** [reply]
Well, electronic equipment can fail. Sometimes, even without a reason. I think you just got bad luck there.

I make several PC-Link cables and on everyone I short pins 4 (DTR - data terminal ready) and 6 (DSR - data set ready) on DB9pin connector, I saw that on schematics Im using and it works without issue. Maybe it isnt neccesary to work but I suggest you to connect the same, I have tested it on dozen of panels from late 90s to late 10s and have hours of programming thru them.

EEPROM memory on newer panels is integrated into main microcontroller processor chip. On older panels, it was seperate 8pin memory located on the back of pcb so it was relatively easy to unlock that panels. Every time you access programming and change/save something, it has to be written in eeprom, and eeprom typically have limited numbers of write cycles, but I think its almost impossible to reach that limit by normal usage. And you shouldn't be able to brick it that easy, with just randomly messing with programming or connecting it to DLS. Also it is not possible to do firmware upgrade on DSC panels.

**Reply by BigHomie90** [reply]
You are right! I asked a friend today who is an alarm installer technician and he told me these things. It happened to him at least twice on old alarm panels that were working fine until a complete power loss. (ac + bad battery). Is it easy for you to draw for me with simple lines how you made the wiring for pc link?Also i want to buy something like this for using dls on my laptop. Is it a good choice? https://www.digitus.info/de/produkte/computer-und-office-zubehoer/computer-zubehoer/usb-komponenten-und-zubehoer/schnittstellen-adapter/da-70156/?PL=en Yeah i know that EEPROM has write limits (as nodemcus, sd cards, etc) but i believe the same thing as you that it can't be that in my case.. "Also it is not possible to do firmware upgrade on DSC panels" - I didnt know that so ok! I ll mess with the new board with less risk now that the upgrade feature is out of the way :P

**Reply by jvitkauskas** [reply]
I've just disassembled a generic "CH340 RS232" adapter https://rarecomponents.com/store/image/cache/data/1590-500x500.jpg you can get on ebay or aliexpress and soldered wires according to this pinout https://pinouts.ru/Electronics/dsc_pclink_conn_pinout.shtml and http://www.chebucto.ns.ca/~am074/PC_Link.html

The only issue was that the adapter is very difficult to disassemble as it is sealed. You can probably use a combination of MAX232 and any USB-TTL adapter instead.

I would advice against purchasing FTDI and Prolific (PL) chips as there are many counterfeits and they block these devices in new drivers. Use either CP2102 or CH340.

EDIT: It seems like the search for "CP2102 RS232" finds an adapter https://i.ebayimg.com/images/g/nrsAAOSwAPVZN9TJ/s-l1600.jpg which has a built-in selector, so you could just select 232 and probably connect it to panel pins directly.

**Reply by kricon** [reply]
> I asked a friend today who is an alarm installer technician and he told me these things. It happened to him at least twice on old alarm panels that were working fine until a complete power loss. (ac + bad battery). Is it easy for you to draw for me with simple lines how you made the wiring for pc link?

What exactly happened to him? Corrupted programming? Its nice to share knowledge here, did he told you any other things worth noting?
For connecting, you did everything good, connection to the panel is fine, but I will suggest bridging pins 4 and 6 on db9 serial connector, as seen in schematic above (DSR and DTR shorted).


@jvitkauskas first link to picture doesnt work. I didnt know CH340/CP2102/FTDI USB-TTL adapters can be used, in combination with MAX232. But I strongly suggest against that kind of configuration, especially when you can buy out-the-box working usb-rs232 for under 3$.

**Reply by jvitkauskas** [reply]
I've fixed that first picture link.

**Comment by BigHomie90** [comment]
He didn't go into detail but i ll ask him more. He said that he encountered the same problem on dsc alarm panels (power series) that they were installed for years. The customers called him for reprogramming the alarm system. When he did all the programming changes the customer wanted, he made a power cycle to the alarm system. At least two of them he remember that they didnt even turn on again.. No clicking, nothing. Yesterday i received my new pc1616 ver. 4.5 EU and i went through local programming , dls , power cycling it many times and still working fine. I did all things as i did to the previous board that it got bricked but the difference is that the new one is still ok! As for the "pc link" i bridged pin 4 and 6 on db9!. 
Also i ordered these (for using dls with my laptop and not serial from dell optiplex) from a local store cause i dont have many options now that we have a lockdown. 

https://www.hellasdigital.gr/electronics/prototyping/adapters/rs-232-serial-to-usb-2.0-ch340-cable-adapter-converter/?sl=en
https://www.hellasdigital.gr/go-create/arduino-shields-and-accessories/db9-rs232-serial-to-terminal-female-adapter-connector/

* Its not CP2102 but still i have some nodeMCU's using CH340g driver and work fine.
